dlp/snippets/inspect_content.py

```diff
@@ -982,6 +984,158 @@ def inspect_image_file_all_infotypes(
+            'VERY_UNLIKELY', 'UNLIKELY', 'POSSIBLE', 'LIKELY', 'VERY_LIKELY'.
+        max_findings: The maximum number of findings to report; 0 = no maximum.
+        timeout: The number of seconds to wait for a response from the API.
+    """
+
+    # This sample also uses threading.Event() to wait for the job to finish.
+    import threading
+
+    # Import the client library.
+    import google.cloud.dlp
+
+    # This sample additionally uses Cloud Pub/Sub to receive results from
+    # potentially long-running operations.
+    import google.cloud.pubsub
+
+    # Instantiate a client.
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Prepare info_types by converting the list of strings into a list of
+    # dictionaries.
+    if not info_types:
+        info_types = ["FIRST_NAME", "LAST_NAME", "EMAIL_ADDRESS"]
+    info_types = [{"name": info_type} for info_type in info_types]
+
+    # Specify how the content should be inspected. Keys which are None may
+    # optionally be omitted entirely.
+    inspect_config = {
+        "info_types": info_types,
+        "min_likelihood": min_likelihood,
+        "limits": {"max_findings_per_request": max_findings},
+        "include_quote": True
+    }
+
+    # Specify the BigQuery table to be inspected.
+    table_reference = {
+        "project_id": bigquery_project,
+        "dataset_id": dataset_id,
+        "table_id": table_id,
+    }
+
+    # Construct a storage_config containing the target BigQuery info.
+    storage_config = {
+        "big_query_options": {
+            "table_reference": table_reference,
+            "rows_limit": 1000,
+            "sample_method": 'RANDOM_START',
+            "identifying_fields": [{"name": "Name"}]
+        }
+    }
+
+    # Tell the API where to send a notification when the job is complete.
+    topic = google.cloud.pubsub.PublisherClient.topic_path(project, topic_id)
+    actions = [{"pub_sub": {"topic": topic}}]
+
+    # Construct the inspect_job, which defines the entire inspect content task.
+    inspect_job = {
+        "inspect_config": inspect_config,
+        "storage_config": storage_config,
+        "actions": actions,
+    }
+
+    # Convert the project id into full resource ids.
+    parent = f"projects/{project}/locations/global"
+
+    # Call the API
+    operation = dlp.create_dlp_job(
+        request={"parent": parent, "inspect_job": inspect_job}
+    )
+    print(f"Inspection operation started: {operation.name}")
+
+    # Create a Pub/Sub client and find the subscription. The subscription is
+    # expected to already be listening to the topic.
+    subscriber = google.cloud.pubsub.SubscriberClient()
+    subscription_path = subscriber.subscription_path(project, subscription_id)
+
+    # Set up a callback to acknowledge a message. This closes around an event
+    # so that it can signal that it is done and the main thread can continue.
+    job_done = threading.Event()
+
+    def callback(message):
```

Comment:
Please add type hints here, similar to the ones on the enclosing function definition.

Reply:
Thank you for your comment. Updated, please check!